docs/source/testing/mocking-responses.mdx

```diff
@@ -0,0 +1,181 @@
+---
+title: Mocking responses
+---
+
+Apollo Kotlin provides two tools to mock responses in tests:
+
+- `TestNetworkTransport`: allows to specify the GraphQL responses returned by `ApolloClient`
+- `MockServer`: a low-level HTTP server that can be used to mock any responses
+
+In most cases, `TestNetworkTransport` will be more straightforward to use in your tests, while `MockServer` can be
+useful to test specific server behavior, such as error cases, custom headers, timeouts, etc.
```

Comment:
Do we want to be opinionated there? I'm personally the kind of person that'd prefer `MockServer` because this touched as little of the production code as possible (basically just changes the `serverUrl`). Maybe we can gather some feedback what's deemed easiest/more flexible, etc... before making a recommendation?

Reply:
Totally agree. Will use your suggestion below.